feat(twofactor): More customization and extended functioniality (#3)
Aiko-IT-Systems/DisCatSharp.Extensions: DisCatSharp.Extensions.Docs/articles/extensions/twofactor_commands/intro.md

To ask a user to submit their two factor code, use the function [RequestTwoFactorAsync()](xref:DisCatSharp.Extensions.TwoFactorCommands.ApplicationCommands.TwoFactorApplicationCommandExtension#DisCatSharp_Extensions_TwoFactorCommands_ApplicationCommands_TwoFactorApplicationCommandExtension_RequestTwoFactorAsync_BaseContext_) on your [BaseContext](xref:DisCatSharp.ApplicationCommands.Context.BaseContext).

> It'll return a [TwoFactorResponse](xref:DisCatSharp.Extensions.TwoFactorCommands.Properties.TwoFactorResponse).

Reviewer: ```suggestion
It returns a [TwoFactorResponse](xref:DisCatSharp.Extensions.TwoFactorCommands.Properties.TwoFactorResponse).
```

```cs

> var tfa_result = ctx.RequestTwoFactorAsync();

Reviewer: ```suggestion
var tfa_result = await tx.RequestTwoFactorAsync();
```

> if (tfa_result != TwoFactorResponse.Result.ValidCode)

Reviewer: ```suggestion
if (tfa_result.Result != TwoFactorResult.ValidCode)
```

{
	// Handle incorrect code
    return;
}

// Do your stuff
```

### Using TwoFactor with Buttons

Using two factor authentication on buttons is pretty similar to slash commands but it'll need a DiscordClient to attach to.

Run [RequestTwoFactorAsync()](xref:DisCatSharp.Extensions.TwoFactorCommands.ApplicationCommands.TwoFactorApplicationCommandExtension#DisCatSharp_Extensions_TwoFactorCommands_ApplicationCommands_TwoFactorApplicationCommandExtension_RequestTwoFactorAsync_BaseContext_) on your [ComponentInteractionCreateEventArgs](xref:DisCatSharp.EventArgs.ComponentInteractionCreateEventArgs) to ask the user for the two factor auth code.

Same deal as for slash commands, it'll return a [TwoFactorResponse](xref:DisCatSharp.Extensions.TwoFactorCommands.Properties.TwoFactorResponse).

```cs
async Task SomeButtonInteraction(DiscordClient sender, ComponentInteractionCreateEventArgs e)
{

>     var tfa_result = e.RequestTwoFactorAsync(sender);

Reviewer: ```suggestion
    var tfa_result = await e.RequestTwoFactorAsync(sender);
```

>     if (tfa_result != TwoFactorResponse.Result.ValidCode)

Reviewer: ```suggestion
    if (tfa_result.Result != TwoFactorResult.ValidCode)
```
